ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/AuthenticationUtils.java

```diff
@@ -53,6 +53,10 @@ public static boolean isPermitted(
                   foundMatchingPermission.set(true);
                 }
               });
+          // exit if a matching permission was found, no need to keep checking
+          if (foundMatchingPermission.get()) {
```

Comment:
Is it possible to create a test that confirms that this exits early?

Reply:
I've had a bit of a look but testing that piece of logic is not straight forward because of the async call. I don't think it's strictly necessary since the main requirement is that the user gets permissions or not, depending on the permissions that are set - and this is already tested.